### PR TITLE
batched jax point cloud and SNR-based CFAR

### LINKED ISSUE #29: `rsp.jax` special implementations (`CFAR`, `CFARCASO`, `PointCloud`) do not have a consistent API
Unlike the rest of `rsp.jax`, `CFAR`, `CFARCASO`, and `PointCloud` are not batched.
- We should either allow batches, or document that we expect the caller to `vmap`.

There's also a few ergonomic issues with the API and documentation:
- we should explicitly call out that `rsp.jax` supports additional features.
- many parameters like `range_resolution` in `PointCloud` should provide a default
- parameters should be better documented, including how they are used and their physical meanings
- axis names are not consistent (`ele` vs `el`, `azi` vs `az`)

### PR DESCRIPTION
**Pull request by huangjuite**
Jax: update from original implementation; follow the CACFAR/CASOCFAR base class
- `PointCloud` processes a batch of cubes, with the angle spectrum argmax reduced to a single reshape + argmax rather than a nested vmap.
- `CFAR` reports a linear-power SNR against a ring-averaged noise floor instead of a z-score.
- `CFARCASO` accumulates its one-sided means from shifted slices, reducing vmap used and improving runtime.
- All batched processes

### CODE REVIEW

File: src/xwr/rsp/jax/aoa.py [hunk 1]
```diff
@@ -1,129 +1,55 @@
...
         range_v = jnp.arange(r_size) * self.range_res
         doppler_v = (jnp.arange(d_size) - d_size // 2) * self.doppler_res
         r_grid, d_grid = jnp.meshgrid(range_v, doppler_v, indexing="ij")
 
-        angle_idx = self.aoa(cube.transpose(3, 0, 1, 2))
-        ang_e = self.ele_angles[angle_idx[:, :, 0]]
-        ang_a = self.azi_angles[angle_idx[:, :, 1]]
-        mask_e = jnp.logical_and(ang_e < self.ele_fov, ang_e > -self.ele_fov)
-        mask_a = jnp.logical_and(ang_a < self.azi_fov, ang_a > -self.azi_fov)
+        angle_idx = self.aoa(cube.transpose(0, 4, 1, 2, 3))
```
Comment (thetianshuhuang) [inline src/xwr/rsp/jax/aoa.py]: Can we use einops here?

File: src/xwr/rsp/jax/aoa.py [hunk 2]
```diff
@@ -1,129 +1,55 @@
...
-            all possible radar points
-        """
-        r_size, d_size = mask.shape
+        cube: Float32[Array, "batch doppler el az range"],
+        mask: Bool[Array, "batch range doppler"],
+    ) -> tuple[
+        Bool[Array, "batch range doppler"],
+        Float32[Array, "batch range doppler 4"],
+    ]:
+        el_angles = self._asarray(self.angle_table(cube.shape[2]), cube)
```
Comment (thetianshuhuang) [inline src/xwr/rsp/jax/aoa.py]: Don't add excess layers of indirection (e.g., _asarray) unless you absolutely have to in order to design the interface. Here it's not doing anything, so just directly use jnp.asarray

File: src/xwr/rsp/jax/spectrum.py [hunk 3]
```diff
@@ -10,231 +9,147 @@
...
-        """1D CFAR CASO.
+        snr: float,
+    ) -> tuple[Bool[Array, "..."], Float[Array, "..."]]:
+        """Run 1D CFAR CASO along `axis` of an arbitrarily batched array.
+
+        The training cells are a contiguous box on each side of the cell under
+        test, so the leading and trailing one-sided means are accumulated
+        directly from shifted slices rather than correlated against a mostly
+        zero kernel. `train` is a static Python int, so the sum unrolls at
+        trace time.
```
Comment (thetianshuhuang) [inline src/xwr/rsp/jax/spectrum.py]: Break paragraphs into bullet points to make them easier to read especially if it's just a loosely related set of implementation notes

**Review by thetianshuhuang — CHANGES_REQUESTED** [review]
Looks good overall with some minor code quality suggestions

File: src/xwr/rsp/jax/spectrum.py [hunk 4]
```diff
@@ -10,231 +9,147 @@
...
-        The user is responsible for applying the desired thresholding.
-        For example, when using a gaussian model, the threshold should be
-        calculated using an inverse normal CDF (e.g. `scipy.stats.norm.isf`):
-
-        ```python
-        cfar = CFAR(guard=(2, 2), window=(4, 4))
-        thresholds = cfar(image)
-        mask = (thresholds > scipy.stats.norm.isf(0.01))
-        ```
+def _integrate(
```
Comment (thetianshuhuang) [inline src/xwr/rsp/jax/spectrum.py]: This is a pretty simple function, it's probably more readable to just inline it

File: src/xwr/rsp/jax/aoa.py (hunk 1 above)
Comment (huangjuite) [reply src/xwr/rsp/jax/aoa.py]: jax doesn't have einops built in; will need an additional library, added a comment on top to show which axis changed, 782c6f10b4054b35953b5182e15ebe5373a16d81

File: src/xwr/rsp/jax/aoa.py (hunk 2 above)
Comment (huangjuite) [reply src/xwr/rsp/jax/aoa.py]: fixed in 782c6f10b4054b35953b5182e15ebe5373a16d81

File: src/xwr/rsp/jax/spectrum.py (hunk 3 above)
Comment (huangjuite) [reply src/xwr/rsp/jax/spectrum.py]: fixed in 782c6f10b4054b35953b5182e15ebe5373a16d81

File: src/xwr/rsp/jax/spectrum.py (hunk 4 above)
Comment (huangjuite) [reply src/xwr/rsp/jax/spectrum.py]: fixed in 782c6f10b4054b35953b5182e15ebe5373a16d81

File: demo/rerun_pc.py [hunk 5]
```diff
@@ -62,10 +62,12 @@ def sig_process(iq):
         cube_rd = rsp.doppler_range(iq)
         cube = rsp.elevation_azimuth(cube_rd)
 
-        rd_mask, sig, snr = cfar.__call__(jnp.abs(cube_rd.squeeze()))
-        pc_mask, pc = radar_pc.__call__(jnp.abs(cube.squeeze()), rd_mask)
+        detection = cfar.__call__(jnp.abs(cube_rd.squeeze()))
```
Comment (thetianshuhuang) [inline demo/rerun_pc.py:65]: Why is there an explicit `__call__` here, why not just `cfar(...)` and so on?

File: src/xwr/rsp/jax/aoa.py [hunk 6]
```diff
@@ -1,131 +1,54 @@
...
-            / (2 * jnp.pi * antenna_spacing)
-        )
-
-    @staticmethod
-    def _argmax_aoa(ang_sptr: Float32[Array, "ele azi"]) -> tuple[Array, ...]:
-        """Argmax for angle of arrival estimation.
+jax.tree_util.register_dataclass(
+    base.DensePoints, data_fields=["mask", "points"], meta_fields=[])
+"""Register [`DensePoints`][xwr.rsp.] as a pytree, so that it can be
+returned from a `jax.jit`-ed function."""
```
Comment (thetianshuhuang) [inline src/xwr/rsp/jax/aoa.py]: This should be a comment, not docstring

File: src/xwr/rsp/jax/spectrum.py [hunk 7]
```diff
@@ -1,245 +1,137 @@
...
 from xwr.rsp import iq_from_iiqq
+from xwr.rsp import spectrum as base
 
 from .rsp import RSPJax
 
 TRSP = TypeVar("TRSP", bound=RSPJax)
 
+jax.tree_util.register_dataclass(
+    base.Detection, data_fields=["mask", "signal", "snr"], meta_fields=[])
+"""Register [`Detection`][xwr.rsp.] as a pytree, so that it can be returned
```
Comment (thetianshuhuang) [inline src/xwr/rsp/jax/spectrum.py]: Same as above, this doesn't need to be a docstring

**Review by thetianshuhuang — APPROVED** [review]
LGTM (with some minor nits)

File: demo/rerun_pc.py (hunk 5 above)
Comment (huangjuite) [reply demo/rerun_pc.py:65]: This is removed in PR #65

File: src/xwr/rsp/jax/aoa.py (hunk 6 above)
Comment (huangjuite) [reply src/xwr/rsp/jax/aoa.py]: fix in e4d45d2a6adb05d9709b273177e270c86a2f250c

File: src/xwr/rsp/jax/spectrum.py (hunk 7 above)
Comment (huangjuite) [reply src/xwr/rsp/jax/spectrum.py]: fix in e4d45d2a6adb05d9709b273177e270c86a2f250c
